### ISSUE TITLE
Refactor: Rewrite test suite

### ISSUE THREAD

**Issue by Evalir** [foundry-rs/foundry#7604]
- [ ] rewrite integration and snapshot tests with a proper framework, improve devex of writing them: https://github.com/foundry-rs/foundry/issues/8389
- [ ] unify `it` and `cli` test folders

Unblocks shell PR [#6569](https://github.com/foundry-rs/foundry/pull/6569).

**Comment by DaniPopes** [comment]
Superseded by https://github.com/foundry-rs/foundry/issues/8389
